public/app/features/dashboard/components/ShareModal/SharePublicDashboard.tsx

```diff
@@ -1,60 +1,85 @@
 import React, { useState, useEffect } from 'react';
 
-import { Button, Field, Switch } from '@grafana/ui';
+import { Button, Field, Switch, Alert } from '@grafana/ui';
 import { notifyApp } from 'app/core/actions';
 import { createErrorNotification, createSuccessNotification } from 'app/core/copy/appNotification';
+import { contextSrv } from 'app/core/services/context_srv';
+import { DashboardModel } from 'app/features/dashboard/state/DashboardModel';
 import { dispatch } from 'app/store/store';
 
 import {
   dashboardCanBePublic,
   getPublicDashboardConfig,
   savePublicDashboardConfig,
+  getDashboard,
   PublicDashboardConfig,
+  DashboardResponse,
 } from './SharePublicDashboardUtils';
 import { ShareModalTabProps } from './types';
 
 interface Props extends ShareModalTabProps {}
 
-// 1. write test for dashboardCanBePublic
-// 2. figure out how to disable the switch
-
 export const SharePublicDashboard = (props: Props) => {
-  const [publicDashboardConfig, setPublicDashboardConfig] = useState<PublicDashboardConfig>({ isPublic: false });
   const dashboardUid = props.dashboard.uid;
+  const orgId = contextSrv.user.orgId;
+  const [publicDashboardConfig, setPublicDashboardConfig] = useState<PublicDashboardConfig>({
+    isPublic: false,
+    publicDashboard: { uid: '', dashboardUid, orgId },
+  });
+
+  const [dashboard, setDashboard] = useState<DashboardModel>({} as DashboardModel);
 
   useEffect(() => {
+    // dashboard model may be stale, so load it ourselves
+    getDashboard(dashboardUid)
+      .then((dashboardResponse: DashboardResponse) => {
+        setDashboard(new DashboardModel(dashboardResponse.dashboard, dashboardResponse.meta));
+      })
+      .catch((err) => {
+        dispatch(notifyApp(createErrorNotification('Failed to retrieve dashboard', err)));
+      });
+
+    // load config
     getPublicDashboardConfig(dashboardUid)
       .then((pdc: PublicDashboardConfig) => {
         setPublicDashboardConfig(pdc);
       })
-      .catch(() => {
-        dispatch(notifyApp(createErrorNotification('Failed to retrieve public dashboard config')));
+      .catch((err) => {
+        dispatch(notifyApp(createErrorNotification('Failed to retrieve public dashboard config', err)));
```

Comment:
If the API returns an http error and a message property this dispatch will be handled by backendSrv so, if you do notifications here was well you will get double notifications